src/main/kotlin/dev/stalla/model/itunes/ItunesCategory.kt

```diff
@@ -1,43 +1,369 @@
 package dev.stalla.model.itunes
 
-import dev.stalla.builder.ItunesCategoryBuilder
-import dev.stalla.builder.validating.ValidatingItunesCategoryBuilder
-import dev.stalla.model.BuilderFactory
-
-/**
- * An [iTunes-style `<category>` tag][https://help.apple.com/itc/podcasts_connect/#/itcb54353390].
- * The same format is also used for Google Play <category> tags, just with a different namespace.
- *
- * @param name The name of the category.
- */
-public sealed class ItunesCategory(public open val name: String) {
-
-    public companion object Factory : BuilderFactory<ItunesCategory, ItunesCategoryBuilder> {
+import dev.stalla.util.findItunesCategory
 
-        /** Returns a builder implementation for building [ItunesCategory] model instances. */
-        @JvmStatic
-        override fun builder(): ItunesCategoryBuilder = ValidatingItunesCategoryBuilder()
-    }
+public sealed class ItunesCategory(public open val name: String) {
 
     /**
      * A simple iTunes-style category, without a nested subcategory:
      *
      * ```
      * <itunes:category text="News" />
      * ```
+     *
+     * Categories are defined in the [Apple Podcasts Categories](https://help.apple.com/itc/podcasts_connect/#/itc9267a2f12).
+     * To define a nested subcategory, use [Simple].
      */
-    public data class Simple(override val name: String) : ItunesCategory(name)
+    public abstract class Simple protected constructor(override val name: String) : ItunesCategory(name)
 
     /**
-     * An iTunes-style category that contains a nested subcategory:
+     * An iTunes-style subcategory that is contained within a parent [Simple]:
      *
      * ```
      * <itunes:category text="News">
      *     <itunes:category text="Tech News" />
      * </itunes:category>
      * ```
      *
-     * @param subcategory The nested [Simple] subcategory.
+     * Categories and their hierarchy are defined in the [Apple Podcasts Categories](https://help.apple.com/itc/podcasts_connect/#/itc9267a2f12).
+     *
+     * @param parent The parent [Nested].
      */
-    public data class Nested(override val name: String, val subcategory: Simple) : ItunesCategory(name)
+    public abstract class Nested protected constructor(override val name: String, public val parent: Simple) : ItunesCategory(name)
+
+    public companion object Factory {
+
+        @JvmStatic
+        public fun from(category: String?): ItunesCategory? = findItunesCategory(category)
+
+        @JvmField
+        public val Arts: Simple = object : Simple("Arts") {}
```

Comment:
The actual category instances are declared directly in the companion object as properties. This way, we have a consistent API experience in Kotlin and Java. Only in the generated Dokka documentation, the instances not quite that obvious to find. Something that will need a bit of explanation in the Readme.

Reply:
Good idea. I'll have to do this another time though – typed down the whole list of these categories once to often today 😓

Reply:
Totally understandable :)